src/main/java/de/tum/cit/aet/artemis/programming/web/ProgrammingExerciseParticipationResource.java

```diff
@@ -314,6 +324,13 @@ public ResponseEntity<Void> resetRepository(@PathVariable Long participationId,
 
         programmingExerciseParticipationService.resetRepository(participation.getVcsRepositoryUri(), sourceURL);
 
+        // ToDo: Circular dependencies currently make it impossible to put this into the ProgrammingExerciseParticipationService
+        // ToDo: Major refactoring would be required
+        if (localVCServletService.isPresent()) {
```

Comment:
I don't think this is the right approach. With LocalVC, there is no need to checkout the repository in this case. 
I also don't think we should expose `localVCServletService` like that. I agree that a larger refactoring would be needed to cleanup the underlying code.

For the particular case I am wondering, if it would not make more sense to use `ContinuousIntegrationTriggerService` and to invoke

`triggerBuild(ProgrammingExerciseParticipation participation)`